packages/common/src/ee/AiAgent/schemas/tools/toolImproveContext.ts

```diff
@@ -0,0 +1,83 @@
+import { z } from 'zod';
+import { AiResultType } from '../../types';
+import { createToolSchema } from '../toolSchemaBuilder';
+
+export const TOOL_IMPROVE_CONTEXT_DESCRIPTION = `
+
+  - **When to use the Improve Context Tool:**
+    - User explicitly corrects your field selections: "No, use net_revenue instead of gross_revenue"
+    - User corrects your terminology interpretation: "Actually, 'total revenue' means revenue_after_tax"
+    - User corrects your logic or calculations: "Conversion rate should be conversions/visits, not conversions/visitors"
+    - User corrects your table/explore choices: "Use the sales_data explore instead of orders table"
+    - User provides better formatting/structure: "I'd structure the examples like this: Category: X, Feedback: Y"
+    - User suggests improvements to prompts or tools: "The prompt needs to be improved, let's add X category"
+    - User clarifies business rules: "Customer count should exclude test accounts"
+    - Your confidence this is a learnable correction (not just a new request) is above 0.7
+
+ - **Correction Categories and Examples:**
+
+    Category: field_selection
+    Feedback: "No, use net_revenue instead of gross_revenue"
+    Generated instruction: "When users ask for revenue, use the net_revenue field instead of gross_revenue"
+
+    Category: terminology
+    Feedback: "Actually, 'total revenue' should always refer to the revenue_after_tax field"
+    Generated instruction: "When users mention 'total revenue', they mean the revenue_after_tax field"
+
+    Category: filter_logic
+    Feedback: "The customer_count should exclude test accounts"
+    Generated instruction: "When calculating customer_count, always exclude test accounts using the filter is_test_account = false"
+
+    Category: calculation
+    Feedback: "Conversion rate should be conversions divided by total visits, not unique visitors"
+    Generated instruction: "When calculating conversion rate, use conversions / total_visits instead of conversions / unique_visitors"
+
+    Category: table_explore_selection
```

Comment:
should it be `table_explore` or just either?

Reply:
:facepalm: yes just explore